src/open_inwoner/openzaak/admin.py

```diff
@@ -90,6 +123,85 @@ class CatalogusConfigAdmin(admin.ModelAdmin):
+            path(
+                "import-catalogus-dump/",
+                self.admin_site.admin_view(self.process_file_view),
+                name="upload_zgw_import_file",
+            ),
+        ]
+        return custom_urls + urls
+
+    @admin.action(description=_("Exporteer naar json export"))
+    def export_catalogus_configs(modeladmin, request, queryset):
+        export = CatalogusConfigExport.from_catalogus_configs(queryset)
+        response = StreamingHttpResponse(
+            export.as_jsonl_iter(),
+            content_type="application/json",
+        )
+        response[
+            "Content-Disposition"
+        ] = 'attachment; filename="zgw-catalogi-export.json"'
+        return response
+
+    def process_file_view(self, request):
+        form = None
+
+        match request.method:
+            case "POST":
+                form = ImportZGWDumpForm(request.POST, request.FILES)
+                if form.is_valid():
+                    storage = PrivateMediaFileSystemStorage()
+                    uploaded_file = request.FILES["zgw_export_file"]
+                    timestamp = datetime.datetime.now().strftime("%Y-%m-%d-%H-%M-%S")
+                    target_file_name = f"zgw_import_dump_{timestamp}.jsonl"
+                    storage.save(target_file_name, uploaded_file)
+
+                    try:
+                        import_result = CatalogusConfigImport.import_from_jsonl_file_in_django_storage(
+                            target_file_name,
+                            storage,
+                        )
+                        self.message_user(
+                            request,
+                            _(
+                                "Successfully processed %(num_rows)d items"
+                                % {"num_rows": import_result.total_rows_processed}
+                            ),
+                            messages.SUCCESS,
+                        )
+
+                        return HttpResponseRedirect(
+                            reverse(
+                                "admin:openzaak_catalogusconfig_changelist",
+                            )
+                        )
+                    except Exception as e:
+                        logger.exception("Unable to process ZGW import")
+                        self.message_user(
+                            request,
+                            _(
+                                "We were unable to process your upload. Please regenerate the file and try again."
+                            ),
+                            messages.ERROR,
+                        )
+                    finally:
+                        storage.delete(target_file_name)
+                else:
+                    self.message_user(
+                        request, _("No file was uploaded"), messages.WARNING
+                    )
+            case _:
+                form = ImportZGWDumpForm()
+
+        return TemplateResponse(
```

Comment:
Should this be indented (as belonging to `case _` where `request.method == GET`?